notebooks/filechecker/argo-file-checker.ipynb

```diff
@@ -0,0 +1,385 @@
+    "        logger.info(\"✅ Results saved to %s\",save_path)\n",
+    "    logger.info(\"✅ Results of first few files %s\",save_path)\n",
+    "    ipy_display(df[:3])\n",
+    "\n",
+    "# Test 1: Check API connection\n",
+    "api_available = test_api_connection()\n",
+    "\n",
+    "if not api_available:\n",
+    "    logger.info(\"\\n API is not accessible. Please check:\")\n",
+    "else:\n",
+    "    logger.info(\"\\n🎉 API is ready for use!\")\n",
+    "# 2. Upload file(s) to test\n",
+    "label = Label(\"(for checking few files) Please upload files to check:\")\n",
+    "upload = FileUpload(accept=\".json,.nc\", multiple=True)\n",
+    "out = Output()\n",
+    "\n",
+    "def on_upload_change(_change: dict[str, object]) -> None:\n",
+    "    \"\"\"To handle file upload and trigger file check.\"\"\"\n",
+    "    with out:\n",
+    "        out.clear_output()\n",
+    "        file_paths = []\n",
+    "        with tempfile.TemporaryDirectory() as tmp_dir:\n",
+    "            for fileinfo in upload.value:\n",
+    "                fname = fileinfo[\"name\"]\n",
+    "\n",
+    "                tmp_path = Path(tmp_dir) / fname\n",
+    "                with Path(tmp_path).open(\"wb\") as f:\n",
+    "                    f.write(fileinfo[\"content\"])\n",
+    "                file_paths.append(tmp_path)\n",
+    "\n",
+    "\n",
+    "            files_data = prepare_files(file_paths)\n",
+    "            res= send_file_check(files_data, DEFAULT_DAC)\n",
+    "            if res is not None:\n",
+    "                show_result(res, save_path=\"file_check_results.csv\")\n",
+    "            else:\n",
+    "                logger.error(\"File check failed, nothing to display\")\n",
+    "\n",
+    "upload.observe(on_upload_change, names=\"value\")\n",
+    "ipy_display(VBox([label, upload, out]))\n",
+    "# Create a text input widget for the deployment folder\n",
+    "\n",
+    "deployment_folder_widget = widgets.Text(\n",
+    "        value=\"//wsl.localhost/Ubuntu/home/vidkri/argo_mount\",  # default value\n",
+    "        placeholder=\"Enter path to deployment folder\",\n",
+    "        description=\"Enter path to deployment folder:\",\n",
+    "        layout=widgets.Layout(width=\"90%\"),\n",
+    "        style={\"description_width\": \"200px\"},\n",
+    "    )\n",
+    "\n",
+    "# Create a button to trigger the check\n",
+    "\n",
+    "run_button = widgets.Button(description=\"Run complete deployment Check\", button_style=\"success\")\n",
+    "\n",
+    "# Define button click handler\n",
+    "def on_run_button_clicked(_b:dict[str, object]) -> None:\n",
+    "    \"\"\"Handle deployment check on button click.\"\"\"\n",
+    "    with output:\n",
+    "        output.clear_output(wait=True)\n",
+    "        deployment_folder = deployment_folder_widget.value.strip()\n",
+    "        if not Path(deployment_folder).exists():\n",
+    "            logger.warning(\"Folder does not exist: %s\" , deployment_folder)\n",
```

Comment:
Something that completely stops your code from fulfilling its intended purpose is definitely not just a warning, that's an error. In fact, this should probably just throw an exception.

Reply:
sorry. agree. It is shown in red now.